src/devices/OneWire/OneWireBus.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Iot.Device.OneWire
+{
+    /// <summary>
+    /// Represents a 1-wire bus.
+    /// </summary>
+    public partial class OneWireBus
+    {
+        internal OneWireBus(string deviceId)
+        {
+            DeviceId = deviceId;
+        }
+
+        /// <summary>
+        /// The 1-wire device id.
+        /// </summary>
+        public string DeviceId { get; }
+
+        /// <summary>
+        /// Enumerate all 1-wire busses in the system.
+        /// </summary>
+        /// <returns>A list of discovered busses.</returns>
+        public static IEnumerable<OneWireBus> EnumerateBuses()
+        {
+            return EnumerateBusesInternal();
+        }
+
+        internal static OneWireDevice CreateDeviceByFamily(OneWireBus bus, string deviceId, DeviceFamily family)
```

Comment:
1. Why is this static if first if of the same type and null doesn't make sense in here?
2. Consider changing to `T CreateDevice<T>(string deviceId) where T : OneWireDevice`
3. Is deviceId actually a string or is this an `int`?

Reply:
1) leftover from controller, will fix
2) ok
3) it's 8bit family, 48bit id and 8bit crc

Reply:
string sounds good in that case